baseapp/abci.go

```diff
@@ -456,14 +456,18 @@ func (app *BaseApp) Commit() abci.ResponseCommit {
 	header := app.deliverState.ctx.BlockHeader()
 	retainHeight := app.GetBlockRetentionHeight(header.Height)
 
+	if app.precommiter != nil {
```

Comment:
@itsdevbear Was curious if this is where you envisioned this being invoked. At this point, your TransientStore will not have been cleared yet.

Reply:
This makes sense to me. If we are passing in the ctx without nuking it, I don't think it really matters where it happens.